samples/nrf9160/serial_lte_modem/src/slm_util.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef SLM_UTIL_
+#define SLM_UTIL_
+
+/**@file slm_util.h
+ *
+ * @brief Utility functions for serial LTE modem
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <ctype.h>
+#include <stdbool.h>
+
+/**
+ * @brief Compare string ignoring case
+ *
+ * @param str1 First string
+ * @param str2 Second string
+ * @param str2_len Length of second string
+ *
+ * @retval true If two commands match, false if not.
+ */
+bool slm_util_cmp(const char *str1, const char *str2, u16_t str2_len);
+
+/**
+ * @brief Compare name of AT command ignoring case
+ *
+ * @param cmd Command string received from UART
+ * @param slm_cmd Propreiatry command supported by SLM
+ * @param slm_cmd_len Length of string to compare
+ *
+ * @retval true If two commands match, false if not.
+ */
+bool slm_util_cmd_cmp(const char *cmd, const char *slm_cmd, u8_t slm_cmd_len);
+
+/**
+ * @brief Detect hexdecimal data type
+ *
+ * @param hex[in] Hex arrary to be encoded
+ * @param hex_len[in] Length of hex array
+ *
+ * @retval true if the input is hexdecimal array, otherwise false
+ */
+bool slm_util_hex_check(const u8_t *hex, u16_t hex_len);
+
+/**
+ * @brief Encode hex array to hexdecimal string (ASCII text)
+ *
+ * @param hex[in] Hex arrary to be encoded
+ * @param hex_len[in] Length of hex array
+ * @param ascii [out] encoded hexdecimal string
```

Comment:
No space before `[out]`, `[in]`, here and all below.

Reply:
Updated